drivers/include/lis3dh.h

```diff
@@ -736,10 +736,8 @@ typedef struct {
  * @brief   Device descriptor for LIS3DH sensors
  */
 typedef struct {
-    spi_t spi;              /**< SPI device the sensor is connected to */
-    spi_clk_t clk;          /**< clock speed of the SPI bus */
-    gpio_t cs;              /**< Chip select pin */
-    int16_t scale;          /**< Current scale setting of the sensor */
+    lis3dh_params_t params; /**< Device initialization parameters */
```

Comment:
We would save some ram if we place params in ROM and use a pointer. The extra indirection should not matter here, the extra few cpu cycles and ROM accesses will not be limiting the performance of the device

Reply:
Some of the parameters are only used by the init function and do not need to be cached in RAM if you are not using a pointer.

Reply:
Agreed, but IMHO this is a more general discussion regarding initialization parameters and how to the handle them. There's also [a wiki page](https://github.com/RIOT-OS/RIOT/wiki/Guide:-Writing-a-device-driver-in-RIOT#device-descriptor-and-parameter-configuration) that explains how to write a device driver. This page doesn't mention the use of a pointer for params.
Other drivers are also not using the params as pointer. Do you think it would make sense to update this in follow-up PR(s) or is it really blocking for you here ?

Reply:
I don't want to block this as there are more pressing issues to address in other parts of the system. However, there is a mention of pointers on the wiki page you linked, excerpt below:

>In many cases it makes sense, to copy the xxx_params data into the device descriptor during initialization. In some cases, it is however better to just link the params data via pointer and only copy selected data.

Reply:
Indeed, you are right!  But in the current state and regarding the discussion here and in #7067, this is confusing I think.